resources/variants/creality_legacy_base_0.2.inst.cfg

```diff
@@ -0,0 +1,12 @@
+[general]
+name = 0.2mm Nozzle
+version = 4
+definition = creality_legacy_base
+
+[metadata]
+setting_version = 9
```

Comment:
```suggestion
setting_version = 10
```

Reply:
Note; this should be for all profiles

Reply:
@nallath I have several people in one of the FB groups where I shared these files telling me they're getting errors when they use the files with "setting_version = 10" in them. If they put the older files in with "setting_version = 9" back, then the errors go away. They tell me this is with Cura 4.3.0.

The error is that the nozzle size is not available in the materials drop down. It just says empty. One person emptied the cache and said it now shows nozzle .4mm but doesn't give a drop down to choose anything else. It seems most of these people are Windows 10 and one is Linux. Mac users haven't reported this problem, and I am working on a Mac myself.

Can you tell me what to look into to troubleshoot this problem please? Also, how should I have learned about the versioning difference so in the future I can get that right before submitting any pull requests.

Reply:
> @nallath I have several people in one of the FB groups where I shared these files telling me they're getting errors when they use the files with "setting_version = 10" in them. If they put the older files in with "setting_version = 9" back, then the errors go away. They tell me this is with Cura 4.3.0.

That's correct. In Cura 4.3 the setting version was 9. In Cura 4.4 it'll be 10. If it's v9, Cura 4.4 will not load it, only 4.3 will. If it's v10, only Cura 4.4 will read it. These profiles are submitted for Master which is next going to be published as Cura 4.4, so it should be v10.

The difference between v9 and v10 is that stacks must now have intent profiles, several profiles have been renamed and that the Alternate Skin Rotation setting can no longer be overridden in profiles. For a complete reference of the renamed profiles, see [this file](https://github.com/Ultimaker/Cura/blob/master/plugins/VersionUpgrade/VersionUpgrade43to44/VersionUpgrade43to44.py). As long as you're not using Alternate Skin Rotations, just updating the `setting_version` to 10 should be fine for you.

> Also, how should I have learned about the versioning difference so in the future I can get that right before submitting any pull requests.

Normally, with pull requests to Master, programmers tend to expect that you've tested your code in Master as well. You would've seen that it didn't load immediately then, and why. We understand that profile designers don't always have the expertise to run Cura from source and then run from the release of 2 months ago, but we haven't really found a way to deal with that yet. Usually it's not really a problem... :confused:

Note that internally the Cura developers haven't agreed yet whether or not this should be merged in. On one hand it would make it easier for profile designers who created custom profiles before May 2019 to get them to use those again. On the other hand we don't want to confuse new users with which definition to use, prevent subjective opinions on which is better, and to open up the possibility of having different definitions for the same printer depending on a person's preferred workflow.

Reply:
@nallath 

> These profiles are submitted for Master which is next going to be published as Cura 4.4, so it should be v10... Normally, with pull requests to Master, programmers tend to expect that you've tested your code in Master as well.

Okay, that makes a lot of sense. Thank you so much.

> Note that internally the Cura developers haven't agreed yet whether or not this should be merged in. 

I completely understand. I want to help where I can and I can appreciate the two sides of the issue here. I think the heart of the matter is that there isn't any way to change all of the settings from within Cura itself so the average user is stuck with a choice of using the old version and missing out on the new features or using the new version and missing out on the print quality they were once able to get. If all of the settings found in the definitions files were available in the GUI, then it would be easy enough for those more familiar with the program to make tutorials and such. Since that isn't a possibility right now, I'm interested in finding a solution for that typical user so they won't need to make that choice. For me, I've figured it out for myself and don't "need" anything further.

So for now, I'll wait on you and the other developers to come to a consensus and if there is something I can do afterwards to contribute, then I'll do so. You might have other ideas I haven't thought of and I'm all ears if you do. Maybe you'd be okay with adding just one extra single printer under the existing manufacturer with a name that makes it clear it is for a generic Creality printer with the old settings and for advanced users. The bed size would have to be edited but that is easy enough in the GUI. Maybe a readme about it can be put in as comments in the start gcode. 

Thanks again for your help and explanations. 

PS Is there a list that tells you which settings for the definitions file are only available in the files and not in the GUI?

Reply:
Yeah a list is [here](https://github.com/Ultimaker/Cura/blob/66005f28f31c343aff697d22ecdbc1d06a6e86c2/resources/definitions/fdmprinter.def.json#L27). We explicitly hide the `machine_settings` category of settings and the `command_line_settings` category. The latter is not very interesting if you have a GUI. The former can be made visible by installing the "Printer Settings" plug-in from fieldOfView.